homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,230 @@
+"""
+Support for the Hive devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hive/
+"""
+import logging
+from datetime import datetime
+
+from homeassistant.components.climate import (ENTITY_ID_FORMAT, ClimateDevice,
+                                              STATE_AUTO, STATE_HEAT,
+                                              STATE_OFF, STATE_ON)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.components.hive import DATA_HIVE
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, hivedevice, discovery_info=None):
+    """Set up Hive climate devices."""
+    session = hass.data.get(DATA_HIVE)
+
+    add_devices([HiveClimateEntity(hass, session, hivedevice)])
+
+
+class HiveClimateEntity(ClimateDevice):
+    """Hive Climate Device."""
+
+    def __init__(self, hass, Session, HiveDevice):
+        """Initialize the Climate device."""
+        self.node_id = HiveDevice["Hive_NodeID"]
+        self.node_name = HiveDevice["Hive_NodeName"]
+        self.device_type = HiveDevice["HA_DeviceType"]
+        self.hass = hass
+        self.session = Session
+
+        if self.device_type == "Heating":
+            set_entity_id = "Hive_Heating"
+            self.session.heating = self.session.core.Heating()
+            self.modes = [STATE_AUTO, STATE_HEAT, STATE_OFF]
+        elif self.device_type == "HotWater":
+            set_entity_id = "Hive_HotWater"
+            self.session.hotwater = self.session.core.Hotwater()
+            self.modes = [STATE_AUTO, STATE_ON, STATE_OFF]
+        if self.node_name is not None:
+            set_entity_id = set_entity_id + "_" \
+                            + self.node_name.replace(" ", "_")
+        self.entity_id = ENTITY_ID_FORMAT.format(set_entity_id.lower())

```

Comment:
I think it would be better to use `generate_entity_id` here, which makes sure that there are no duplicates.

Reply:
We shouldn't be manually assigning IDs at all. Just set the entity name.

Reply:
this ID setting was to aid those with multizone heating identify each of their zones and keep things constant
what would happen to the `entity_id` if one time HA started and zone1 was added first and zone2 second, but then next time zone2 was added first and zone1 was added second?
wouldn't it mess up the automations and other stuff?

Reply:
I have pushed a new version which does not `set entity_id` but I am sure when testing this with a multizone heating setup this caused problems which was why I put this code in, in the first place